tiledb/api/c_api/group/group_api_external.h

```diff
@@ -373,6 +373,40 @@ TILEDB_EXPORT capi_return_t tiledb_group_add_member(
     const uint8_t relative,
     const char* name) TILEDB_NOEXCEPT;
 
+/**
+ * Add a member with a known type to a group
+ * The caller should make sure the correct type is provided for the member being
+ * added as this API will not check it for correctness in favor of efficiency
+ * and results can be undefined otherwise.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_group_t* group;
+ * tiledb_group_alloc(ctx, "s3://tiledb_bucket/my_group", &group);
+ * tiledb_group_open(ctx, group, TILEDB_WRITE);
+ * tiledb_group_add_member(ctx, group, "s3://tiledb_bucket/my_array",
+ * TILEDB_ARRAY); tiledb_group_add_member(ctx, group,
+ * "s3://tiledb_bucket/my_group_2", TILEDB_GROUP);
```

Comment:
* The code needs formatting.
* We should call the new function.
* We should close the group at the end (it's important otherwise the changes will not be persisted).